helix-core/src/chars.rs

```diff
@@ -81,8 +81,26 @@ pub fn char_is_punctuation(ch: char) -> bool {
 }
 
 #[inline]
-pub fn char_is_word(ch: char) -> bool {
-    ch.is_alphanumeric() || ch == '_'
+pub fn char_is_word(ch: char) -> Option<u8> {
```

Comment:
let's have this `char_is_word` function still return a `bool` and introduce another function that classifies the subcagetory. Then `char_is_word` can just check `is_some()` and the callsites that were updated to check `is_some`/`is_none` can be reset